src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -153,7 +153,33 @@ public AzureAppConfigurationOptions Select(string keyFilter, string labelFilter
                 _kvSelectors.Add(new KeyValueSelector
                 {
                     KeyFilter = keyFilter,
-                    LabelFilter = labelFilter
+                    LabelFilter = labelFilter,
+                    SnapshotName = string.Empty
+                });
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Specify a snapshot and include its captured key-values in the configuration provider.
+        /// <see cref="SelectSnapshot"/> can be called multiple times to include key-values from multiple snapshots.
+        /// </summary>
+        /// <param name="name">The name of the snapshot in Azure App Configuration.</param>
+        public AzureAppConfigurationOptions SelectSnapshot(string name)
```

Comment:
Please add a check for the composition_type.key before loading a snapshot; throw otherwise.